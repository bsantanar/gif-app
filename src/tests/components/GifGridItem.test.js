import React from 'react';

import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Title'
    const url = 'http://localhost:8000'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)
    
    test('should show component', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should contains title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    
    test('should contain img with url and alt', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    
    test('should ', () => {
        const div = wrapper.find('div')

        expect(div.hasClass('animate__fadeInUp')).toBe(true)
    })
    

})

