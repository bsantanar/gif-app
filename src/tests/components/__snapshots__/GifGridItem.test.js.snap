// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should show component 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Title"
    src="http://localhost:8000"
  />
  <p>
    Title
  </p>
</div>
`;
